pkg/sdk/session.go

```diff
@@ -0,0 +1,106 @@
+package sdk
+
+import (
+	"context"
+	"log"
+	"net/url"
+
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+// sessionSuppliers is the structure that represents a session's end block height
+// and its matching suppliers.
+type sessionSuppliers struct {
+	// SessionEndBlockHeight is the session's end block height that helps
+	// determine if the session is still valid without looking into SupplierEndpoints slice.
+	SessionEndBlockHeight int64
+	SuppliersEndpoints    []*SupplierEndpoint
+}
+
+// SupplierEndpoint is the structure that represents a supplier's endpoint
+// augmented with the session's header and the supplier's address for easy
+// access to the needed information when sending a relay request.
+type SupplierEndpoint struct {
+	Url             *url.URL
+	RpcType         sharedtypes.RPCType
+	SupplierAddress string
+	Header          *sessiontypes.SessionHeader
+}
+
+// GetCurrentSession gets the current session for the given service
+// It returns the current state if it exists and is still valid, otherwise it
+// queries for the latest session, constructs the SessionSuppliers, caches and
+// returns its SupplierEndpoints.
+func (sdk *poktrollSDK) GetCurrentSession(
```

Comment:
The function name is a bit inappropriate. I feel like it should be called `GetSessionSupplierEdnpoints` and update the comment to say that it `flattens the endpoints from all suppliers in the session`